rsocket-core/src/main/java/io/rsocket/lease/LeaseHandler.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright 2015-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.rsocket.lease;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufAllocator;
+import io.rsocket.Availability;
+import io.rsocket.exceptions.MissingLeaseException;
+import io.rsocket.frame.LeaseFrameFlyweight;
+import java.util.Objects;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import javax.annotation.Nullable;
+import reactor.core.Disposable;
+import reactor.core.publisher.Mono;
+import reactor.core.publisher.MonoProcessor;
+
+public class LeaseHandler {
+  private final MonoProcessor<Void> onClose = MonoProcessor.create();
```

Comment:
Until ClientServerInputMultiplexer is fixed, they are tied because lease frame has to be received in both requester and responder to update requester lease.

Reply:
Only the requester needs to receive a lease frame - the responder sends the frame to requester - why do they both need to receive it?

Reply:
Ok  I will fix clientServerMultiplexer